src/core/Rule.ml

```diff
@@ -275,6 +275,31 @@ and extract_transform = NoTransform | Unquote | ConcatJsonArray
 *)
 and extract_reduction = Separate | Concat [@@deriving show]
 
+(*****************************************************************************)
+(* secrets mode *)
+(*****************************************************************************)
+
+type request_spec = {
+  url : Uri.t;
+  meth : [`GET | `POST];
+  headers : (string * string) list
+}
+[@@deriving show]
+
+type response_sepc = {
```

Comment:
typo, spec no sepc.
Also I would remove those _spec prefix (I have a PR where I removed it from extract; I kept the
taint_spec because there's code depending on this name type, and I was lazy, but I think it would
be simpler to remove those _spec suffix from the start here.

Reply:
Removed _spec